### PR TITLE
feat: Tabs add slot to support extra position

### PR DESCRIPTION
**Pull request by jesse3mh9a**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/24522#issue-626162474

### 💡 Background and solution
https://github.com/react-component/tabs/pull/286#issue-437512991

[DEMO](https://tabs-g7aoi7ju9.vercel.app/?path=/story/rc-tabs--extra)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |         Tabs `tabBarExtraSlot` now support extra position.    |
| 🇨🇳 Chinese |      `Tabs` 可自定义`tabBarExtraContent` 的渲染位置。     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/tabs/index.en-US.md](https://github.com/jesse3mh9a/ant-design/blob/tabs-extra-slot/components/tabs/index.en-US.md)

### CODE REVIEW

**Comment by jesse3mh9a** [comment]
[Demo](https://tabs-go3ahrhnm.vercel.app/?path=/story/rc-tabs--extra)

**Comment by afc163** [comment]
https://github.com/react-component/tabs/releases/tag/v11.6.0

**Comment by jesse3mh9a** [comment]
@afc163  package, api doc, and demo have been updated. ✅

**Comment by afc163** [comment]
rebase feature

**Comment by jesse3mh9a** [comment]
all passed @afc163

File: components/tabs/index.zh-CN.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ Ant Design 依次提供了三级选项卡，分别用于不同的场景。
 | hideAdd | 是否隐藏加号图标，在 `type="editable-card"` 时有效 | boolean | false |  |
 | size | 大小，提供 `large` `default` 和 `small` 三种大小 | string | `default` |  |
 | centered | 标签居中展示 | boolean | false | 4.4.0 |
-| tabBarExtraContent | tab bar 上额外的元素 | ReactNode | - |  |
+| tabBarExtraContent | tab bar 上额外的元素 | `ReactNode` \| `{left?: ReactNode, right?: ReactNode}` | - |  |
```
Comment (Kermit-Xuan) [inline components/tabs/index.zh-CN.md:36]: 前后的 ReactNode 最好保持一致，前面的是不是应该去掉 `?

Comment (xrkffgg) [reply components/tabs/index.zh-CN.md:36]:
```suggestion
| tabBarExtraContent | tab bar 上额外的元素 | ReactNode \| {left?: ReactNode, right?: ReactNode} | - |  |
```

File: components/tabs/demo/extra.md [hunk 2]
```diff
@@ -7,31 +7,81 @@ title:
 
 ## zh-CN
 
-可以在页签右边添加附加操作。
+可以在页签右边或左边甚至两边添加附加操作。
```
Comment (xrkffgg) [inline components/tabs/demo/extra.md:10]:
```suggestion
可以在页签两边添加附加操作。
```

Comment (jesse3mh9a) [reply components/tabs/demo/extra.md]: 👌

**Review by kerm1it — APPROVED** [review]

**Pull request merged**